### PR TITLE
feat(docs): Add model config spec

### LINKED ISSUE #3: [enhancement] Design v1 Model Config Specification
We should define the specification for mediaType `application/vnd.caicloud.model.config.v1+json`. 

We can refer to https://github.com/opencontainers/image-spec/blob/master/config.md

/priority p0

### PR DESCRIPTION
**Pull request by gaocegege**
Signed-off-by: Ce Gao <gaoce@caicloud.io>

**What this PR does / why we need it**:

Add model config v1alpha1 specification, compared with the internal manifest definition:

- Add extensibility for dataset and training. Currently we only support git repo but keep the extensibility for other sources.
- Remove resource limits in the manifest

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #3 

Reference to #

**Special notes for your reviewer**:

/cc @ddysher @codeflitting 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by codeflitting** [comment]
LGTM

/cc @simon-cj

**Comment by simon-cj** [comment]
LGTM

File: docs/spec-v1alpha1.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+  - [Properties](#properties)
+  - [Example](#example)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# OCI Model Configuration Specification
+
+An OCI Model is a Machine Learning/Deep Learning model. This specification outlines the JSON format describing models for use and execution tool and its relationship to filesystem changesets.
+
+This section defines the `application/vnd.caicloud.model.config.v1alpha1+json` media type.
```
Comment (ddysher) [inline docs/spec-v1alpha1.md:18]: Should we get rid of `caicloud` from the beginning?

File: docs/spec-v1alpha1.md [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+    Size of the tensor.
+
+### [GitRepo](https://kubernetes.io/docs/concepts/storage/volumes/#gitrepo)
+
+GitRepo is the Git repository.
+
+- **repository** string, OPTIONAL
+
+    Repository of the training scripts
```
Comment (ddysher) [inline docs/spec-v1alpha1.md:50]: Repository can be used for data source as well, is it too restricted to describe it as `Repository of the training scripts`

File: docs/spec-v1alpha1.md [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+
+- **description** string, OPTIONAL
+
+    Description of the training script
+
+## Properties
+
+Note: Any OPTIONAL field MAY also be set to null, which is equivalent to being absent.
+
+- **created** string, OPTIONAL
```
Comment (ddysher) [inline docs/spec-v1alpha1.md:64]: is this a well-established field in other specs?  It's not intuitive to know what `created` mean by just looking at the field name..

File: docs/spec-v1alpha1.md [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+
+    Description of the model
+
+- **tags** object, OPTIONAL
+
+    Tags of the model
+
+- **labels** object, OPTIONAL
+
+    The field contains arbitrary metadata for the container. This property MUST use the annotation rules.
```
Comment (ddysher) [inline docs/spec-v1alpha1.md:82]: What is `annotation rules`?

File: docs/spec-v1alpha1.md [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+
+- **framework** string, OPTIONAL
+
+    The framework of the model (e.g. TensorFlow)
+
+- **format** string, REQUIRED
+
+    The format of the model (savedmodel, onnx and so on)
```
Comment (ddysher) [inline docs/spec-v1alpha1.md:90]:
```suggestion
    The format of the model (SavedModel, ONNX and so on)
```

nit: use upper case

File: docs/spec-v1alpha1.md [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+
+- **size** int, REQUIRED
```
Comment (ddysher) [inline docs/spec-v1alpha1.md:92]: Is there any unit requirement for `size`? e.g. `10Gi` or `10G` or others?

File: docs/spec-v1alpha1.md (hunk 2 above)
Comment (gaocegege) [reply docs/spec-v1alpha1.md:50]: SGTM, will update.

File: docs/spec-v1alpha1.md (hunk 3 above)
Comment (gaocegege) [reply docs/spec-v1alpha1.md:64]: It is a well-established field in OCI image spec

File: docs/spec-v1alpha1.md (hunk 4 above)
Comment (gaocegege) [reply docs/spec-v1alpha1.md:82]: Should have a link here, will add it.

File: docs/spec-v1alpha1.md (hunk 6 above)
Comment (gaocegege) [reply docs/spec-v1alpha1.md:92]: The size is in defined in containerd github.com/containerd/containerd/content/content.go

We convert it to human-readable size:

```
// byteCountBinary produces a human-readable file size
func byteCountBinary(b int64) string {
	const unit = 1024
	if b < unit {
		return fmt.Sprintf("%d B", b)
	}
	div, exp := int64(unit), 0
	for n := b / unit; n >= unit; n /= unit {
		div *= unit
		exp++
	}
	return fmt.Sprintf("%.1f %ciB", float64(b)/float64(div), "KMGTPE"[exp])
}
```

Do you think we should keep a string to use the human-readable representation?

Comment (ddysher) [reply docs/spec-v1alpha1.md:92]: what do you mean by ` keep a string to use the human-readable representation?`

Comment (gaocegege) [reply docs/spec-v1alpha1.md:92]: Now I propose to use `**size** int, REQUIRED`, do you want to use `**size** string, REQUIRED` to store `8Gi` or `8Mi`?

Comment (ddysher) [reply docs/spec-v1alpha1.md:92]: If this is mostly generated automatically, then using `int` would be fine.

File: docs/spec-v1alpha1.md (hunk 1 above)
Comment (gaocegege) [reply docs/spec-v1alpha1.md:18]: Then will we use kubeflow? Or a new org name?

File: docs/spec-v1alpha1.md [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+   "tags": [
+       "cv"
+   ],
+   "labels": {
+       "tensorflow.version": "2.0.0"
+   },
+   "framework": "TensorFlow",
+   "format": "SavedModel",
+   "size": 9223372036854775807,
+   "metrics": [
```
Comment (codeflitting) [inline docs/spec-v1alpha1.md:168]: Does it parse automatically here？
If not, then we don't have a place to put our custom metrics

Comment (gaocegege) [reply docs/spec-v1alpha1.md:168]: It should be provided by users (Users may be the CLI user, or our model registry)

Our custom metrics will be provided by the model registry to ormb.

**Comment by codeflitting** [comment]
LGTM

File: docs/spec-v1alpha1.md [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+- **dataset** object, OPTIONAL
+
+    - **git** *GitRepo*, OPTIONAL
+
+        Git repository of the dataset
+
+## Example
+
+Here is an example model configuration JSON document:
+
```
Comment (codeflitting) [inline docs/spec-v1alpha1.md:153]: We should add a subcommand later to print the spec info

File: docs/spec-v1alpha1.md (hunk 3 above)
Comment (ddysher) [reply docs/spec-v1alpha1.md:64]: ok

File: docs/spec-v1alpha1.md (hunk 1 above)
Comment (ddysher) [reply docs/spec-v1alpha1.md:18]: I think we should have a new org. Let's change this after we create the new org.

File: docs/spec-v1alpha1.md (hunk 8 above)
Comment (gaocegege) [reply docs/spec-v1alpha1.md:153]: Will do in #13

**Pull request merged**